app/plugin_key_value_store.go

```diff
@@ -57,6 +65,17 @@ func (a *App) GetPluginKey(pluginId string, key string) ([]byte, *model.AppError
 
 	kv := result.Data.(*model.PluginKeyValue)
 
+	// If we are here that means we are using old hashed key. Remove it and insert the new key without hashing it
+	_ = <-a.Srv.Store.Plugin().Delete(pluginId, getKeyHash(key))
+
+	// Insert the key without hashing
```

Comment:
If we keep this code, I think we can just rely on `SetPluginKeyWithExpiry` deleting the old key, right?

Reply:
Correct